internal/security/fileprovider/init.go

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package fileprovider
+
+import (
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// Global variables
+var Configuration *ConfigurationStruct
+var LoggingClient logger.LoggingClient
+
+func Retry(useRegistry bool, useProfile string, timeout int, wait *sync.WaitGroup, ch chan error) {
+	until := time.Now().Add(time.Millisecond * time.Duration(timeout))
+	for time.Now().Before(until) {
+		var err error
+		// When looping, only handle configuration if it hasn't already been set.
+		if Configuration == nil {
+			Configuration, err = initializeConfiguration(useRegistry, useProfile)
+			if err != nil {
+				ch <- err
+				if !useRegistry {
+					// Error occurred when attempting to read from local filesystem. Fail fast.
+					close(ch)
+					wait.Done()
+					return
+				}
+			} else {
+				// Setup Logging
+				logTarget := setLoggingTarget()
+				LoggingClient = logger.NewClient(internal.SecurityProxySetupServiceKey, Configuration.Logging.EnableRemote, logTarget, Configuration.Writable.LogLevel)
+			}
+		}
+		// This seems a bit artificial here due to lack of additional service requirements
+		// but conforms to the pattern found in other edgex-go services.
+		if Configuration != nil {
+			break
+		}
+		time.Sleep(time.Second * time.Duration(1))
+	}
+	close(ch)
+	wait.Done()
+
+	return
+}
+
+func initializeConfiguration(useRegistry bool, useProfile string) (*ConfigurationStruct, error) {
+	// We currently have to load configuration from filesystem first in order to obtain Registry Host/Port
+	configuration := &ConfigurationStruct{}
+	err := config.LoadFromFile(useProfile, configuration)
```

Comment:
This line broken by commit https://github.com/edgexfoundry/edgex-go/commit/f758cd6862d8e6cc426c71a8990c23b386da041b#diff-4f3fa7b8b914d62f80a77f6d084f1c8aR43 which changed public API of 'config' module.